src/Microsoft.Azure.SignalR.Common/ClaimsUtility.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using Microsoft.Azure.SignalR.Protocol;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal static class ClaimsUtility
+    {
+        private const string DefaultAuthenticationType = "Bearer";
+        private static readonly string[] SystemClaims =
+        {
+            "aud", // Audience claim, used by service to make sure token is matched with target resource.
+            "exp", // Expiration time claims. A token is valid only before its expiration time.
+            "iat", // Issued At claim. Added by default. It is not validated by service.
+            "nbf"  // Not Before claim. Added by default. It is not validated by service.
+        };
+
+        private static readonly ClaimsIdentity DefaultClaimsIdentity = new ClaimsIdentity();
+        private static readonly ClaimsPrincipal EmptyPrincipal = new ClaimsPrincipal(DefaultClaimsIdentity);
+        private static readonly string DefaultNameClaimType = DefaultClaimsIdentity.NameClaimType;
+        private static readonly string DefaultRoleClaimType = DefaultClaimsIdentity.RoleClaimType;
+
+        public static IEnumerable<Claim> BuildJwtClaims(ClaimsPrincipal user, string userId, Func<IEnumerable<Claim>> claimsProvider)
+        {
+            if (userId != null)
+            {
+                yield return new Claim(Constants.ClaimType.UserId, userId);
+            }
+
+            var authenticationType = user?.Identity?.AuthenticationType;
+
+            // No need to pass it when the authentication type is Bearer
+            if (authenticationType != null && authenticationType != DefaultAuthenticationType)
+            {
+                yield return new Claim(Constants.ClaimType.AuthenticationType, authenticationType);
+            }
+
+            // return custom NameClaimType and RoleClaimType
+            if (user?.Identity is ClaimsIdentity identity)
```

Comment:
I think you can have multiple identities but I'm OK ignoring this until we find that it's required.

Reply:
Add a comment for this